src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -178,5 +181,42 @@ public bool ManualSettingsOnly
 
             return useProxy;
         }
+
+        public bool AutoDetectScriptSuccess()
+        {
+            // Only check WinHttpDetectAutoProxyConfigUrl if auto detect is set.
+            if (!AutoDetect) return false;
+
+            bool doAutoDetect = false;
+            int currentTimeStamp = Environment.TickCount;
+
+            // Environment.TickCount will increment from zero to Int32.MaxValue for approximately 24.9 days, then jump to
+            // Int32.MinValue. When that happens, we may skip calling WinHttpDetectAutoProxyConfigUrl for one request which we
+            // otherwise will call. Since _previousTimeStampTicks will get updated in that request, subsequent requests in
+            // next 24.9 days period will behave correctly.
+            doAutoDetect = currentTimeStamp - _previousTimeStampTicks > TimerPeriodInMilliseconds;
+            _previousTimeStampTicks = currentTimeStamp;
+
+            if (doAutoDetect)
```

Comment:
I don't understand this logic in this method.

The whole point of it is that if "AutoDetect" is true (in IE settings), there might be a PAC file on the network. But if there isn't, then we want the calling code to know that and not call WinHttpGetProxyForUrl().  So, this method is really trying to cache a definitive "NO" answer from WinHttpDetectAutoProxyConfigUrl().

So, shouldn't we be caching the previous return value of calling this p/invoke along with a timestamp.  And then the algorithm is that if the previous call was NO (false) but the timestamp is still fresh, then we return false.  But if the timestamp is old then we call WinHttpDetectAutoProxyConfigUrl() again and reset the answer with a new timestamp.  

And so, calculating a new timestamp should only be done inside this if conditional that is required here.